src/de/mod.rs

```diff
@@ -653,6 +653,17 @@ mod tests {
         assert!(from_str::<[i32; 2]>("[0, 1,]").is_err());
     }
 
+    #[test]
+    fn tuple_variant() {
+        #[derive(Debug, Deserialize, PartialEq)]
+        enum TupleVariant {
+            Add(i32, i32),
+            Sub(i32, i32),
```

Comment:
Could we test different argument length, like 

```suggestion
            Exit(),
            Suare(i32),
            Add(i32, i32),
```

Reply:
Nice idea. The one-element tuple has in fact a slightly different syntax (square brackets are removed). Tests now document this (and check compatibility with `serde_json`).